sharktank/sharktank/models/llama/tools/import_quark_dataset.py

```diff
@@ -0,0 +1,366 @@
+):
+    """Convert layernames for non quantized tensors and add them to the updated_tensors dict"""
+    for sub in ["input_layernorm", "post_attention_layernorm"]:
+        sub_name = layer_name + "." + sub
+        new_name = hf_to_gguf(sub_name) + ".weight"
+        single_replace(quant_theta, sub_name, new_name, updated_tensors)
+    kv_cache_scale = (
+        quant_theta(layer_name).tensor("kv_cache_scaling_factor").as_torch()
+    )
+    layer_idx = layer_name.split(".")[-1]
+    new_name = f"blk.{layer_idx}.kv_cache"
+    kv_cache_scale = DefaultPrimitiveTensor(
+        name=new_name + ".kv_cache_scaling_factor", data=kv_cache_scale
+    )
+    updated_tensors[new_name] = kv_cache_scale
+
+
+def single_replace(
+    quant_theta: Theta,
+    layer_name: str,
+    gguf_name: str,
+    updated_tensors: dict[str, InferenceTensor],
+):
+    data = quant_theta(layer_name).tensor("weight").as_torch()
+    updated_tensors[gguf_name] = DefaultPrimitiveTensor(name=gguf_name, data=data)
+
+
+def main(argv):
+    from sharktank.utils import cli
+
+    parser = cli.create_parser()
+    cli.add_output_dataset_options(parser)
+    parser.add_argument(
+        "--config-json", type=Path, required=True, help="Path to the config.json file"
+    )
+    parser.add_argument(
+        "--params",
+        type=Path,
+        default=Path("params.safetensors"),
+        help="Parameter file name, relative to config.json",
+    )
+    parser.add_argument(
+        "--model-base",
+        type=str,
+        default="7b",
+        help="Base model to use for split sizes to decompose the qkv tensor. Default is 7b, 70b is also supported.",
+        choices=["7b", "70b"],
+    )
+    args = cli.parse(parser, args=argv)
+
+    config_json_path: Path = args.config_json
+    params_path: Path = args.params
+    # TODO: find a way to get this programatically so we don't have to flag for it
+    split_sizes = [4096, 4096, 4096] if args.model_base == "7b" else [8192, 1024, 1024]
+    num_layers = 32 if args.model_base == "7b" else 80
+
+    # Construct the pre-transform dataset.
+    dataset_props = _get_dataset_props(_load_json(config_json_path))
+    with safetensors.safe_open(params_path, framework="pt", device="cpu") as st:
+        quant_theta = _load_theta(st)
+    ds = Dataset(dataset_props, quant_theta)
+
+    # Convert hyperparams to gguf format
+    updated_properties = convert_hf_hparams_to_gguf(ds.properties)
+
+    updated_tensors: dict[str, InferenceTensor] = {}
+    model_layers = [f"model.layers.{i}" for i in range(num_layers)]
+
+    sub_layers = [
+        "mlp.gate_proj",
+        "mlp.down_proj",
+        "mlp.up_proj",
+        "self_attn.o_proj",
+        "self_attn.qkv",
+    ]
+    for layer in model_layers:
+        for sub in sub_layers:
+            layer_name = layer + "." + sub
+            apply_per_layer_quant(
+                quant_theta, layer_name, updated_tensors, split_sizes=split_sizes
```

Comment:
missing n_head arg

Reply:
nice catch, the default is correct for 7b but will be a problem for 70